src/libcore/result.rs

```diff
@@ -792,6 +792,39 @@ impl<T: fmt::Debug, E> Result<T, E> {
     }
 }
 
+impl<T: Default, E> Result<T, E> {
+    /// Returns the contained value or a default
+    ///
+    /// Consumes the `self` argument then, if `Ok`, returns the contained
+    /// value, otherwise if `Err`, returns the default value for that
+    /// type.
+    ///
+    /// # Examples
+    ///
+    /// Convert a string to an integer, turning poorly-formed strings
+    /// into 0 (the default value for integers). `parse` converts
+    /// a string to any other type that implements `FromStr`, returning an
+    /// `Err` on error.
+    ///
+    /// ```
+    /// let good_year_from_input = "1909";
+    /// let bad_year_from_input = "190blarg";
+    /// let good_year = good_year_from_input.parse().unwrap_or_default();
+    /// let bad_year = bad_year_from_input.parse().unwrap_or_default();
+    ///
+    /// assert_eq!(1909, good_year);
+    /// assert_eq!(0, bad_year);
+    /// ```
+    #[inline]
+    #[stable(feature = "rust1", since = "1.0.0")]
```

Comment:
instruction: Should be unstable, no version number. Pick a new feature name, for example "result_unwrap_or_default". Use `issue = "0"` in the unstable tag; an issue can be opened and filled in when this PR has been cleared for inclusion.